geo/src/types.rs

```diff
@@ -37,6 +37,16 @@ impl<F: GeoFloat> Closest<F> {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum GeoError {
+    MissingOverlap,
+    SegmentNotFoundInActiveSet(usize),
+    SegmentAlreadyFoundInActiveSet(usize),
+    ExpectedNonemptyEvents,
+    NotEulierian(usize),
+    Unreachable(&'static str),
```

Comment:
Just to put my cards on the table, I think writing panics into our code is reasonable if the programmer truly expects the code to be unreachable, and that trying to turn every single panic into a runtime error is probably an anti-pattern.

My preference is that this be only the set of run time errors we're actually encountering. Is it the case that you're hitting all of these? 

If we are actually encountering one of the "Unreachable" cases, it's not unreachable. Can you think of a better name that might help people understand what's going on?

Reply:
Yeah, unfortunately I was able to trigger the unreachable code (see the 3rd case in the spoiler of [this comment](https://github.com/georust/geo/issues/976#issuecomment-1615055706)).

Again, I agree here as well. We should probably pick a better name here. I'll look into that. I wasn't diving too deep into that code yet since I just wanted to have some kind of parity to the existing panicing implementation, but I guess now's a good time for that.